lib/web/apiserver.go

```diff
@@ -612,13 +614,18 @@ func NewHandler(cfg Config, opts ...HandlerOption) (*APIHandler, error) {
 		h.nodeWatcher = cfg.NodeWatcher
 	}
 
-	routingHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		// ensure security headers are set for all responses
-		httplib.SetDefaultSecurityHeaders(w.Header())
-
-		// request is going to the API?
-		if strings.HasPrefix(r.URL.Path, apiPrefix) {
-			http.StripPrefix(apiPrefix, h).ServeHTTP(w, r)
+	notFoundRoutingHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Request is going to the API?
+		// If no routes were matched, it could be because certain paths don't expect any version prefixes.
+		// Only prefix "v1" will be stripped to preserve the legacy behavior before introducing v1 + N.
+		if matches := apiPrefixRegex.FindStringSubmatch(r.URL.Path); matches != nil && len(matches) > 1 {
+			versionNum := matches[1]
+			if versionNum == "1" {
+				// Try path matching again with the stripped prefix.
```

Comment:
i can't seem to follow this. wouldn't this catch any version prefixed api path, and if its not v1, then it replies with not found? where does it send the other version number requests to? how does it break out of this

Reply:
so this entire block of code is only ran if the first attempt at routing match failed 

i also added more comments to this block on the newest changes that may explain it more?

Reply:
that makes sense, thats why its called notFound handler now. ok i like it